### DISCUSSION TITLE
Does Keycloak Always Query AD server While Validating User Tokens in Domain Extensions?

### DISCUSSION THREAD

**Discussion by tishan18927** [keycloak/keycloak#29127] [Q&A]
I have created a REST domain extension as provided by `testsuite/integration-arquillian/servers/auth-server/services/testsuite-providers/src/main/java/org/keycloak/testsuite/domainextension/rest/ExampleRestResource.java`. I want to know if I use `new AppAuthManager.BearerTokenAuthenticator(session).authenticate()` as demonstrated in the example, will it always query LDAP server each time a LDAP user is accessing this REST resource because of the `org.keycloak.services.managers.AuthenticationManager#verifyIdentityToken`.

Furthermore, I do not want to check any roles or permissions for this specific use case but only the validity of the token. Given that information, is there any other provider I can use to validate the token. `org.keycloak.protocol.oidc.AccessTokenIntrospectionProvider#introspect` seems to be a good candidate but since it is returning a `Response` type object, it is hard to deal with

**Comment by codespearhead** [comment]
Whether or not it'll query the LDAP server depends on how user federation is configured and especially on cache settings.

I suggest you keep using that method, given that you stated you don't need role checks or to call any role-check methods or methods after authentication.

https://github.com/keycloak/keycloak/blob/99cb437dc76d9cec8ed5bd122cfe48cc04d0e3d7/testsuite/integration-arquillian/servers/auth-server/services/testsuite-providers/src/main/java/org/keycloak/testsuite/domainextension/rest/ExampleRestResource.java#L35
